src/test/perl/PostgreSQL/Test/TdeCluster.pm

```diff
@@ -0,0 +1,228 @@
+
+
+package PostgreSQL::Test::TdeCluster;
+
+use parent 'PostgreSQL::Test::Cluster';
+
+use strict;
+use warnings FATAL => 'all';
+
+use PostgreSQL::Test::RecursiveCopy ();
+use PostgreSQL::Test::Utils ();
+
+sub init
+{
+	my ($self, %params) = @_;
+
+	$self->SUPER::init(%params);
+
+	# TODO: This will not be enough for tests that add other preload libraries
+	$self->append_conf('postgresql.conf', 'shared_preload_libraries = pg_tde');
+	$self->append_conf('postgresql.conf', 'pg_tde.wal_encrypt = on');
+
+	$self->_setup_keyfile;
+
+	unless (-e $self->pg_tde_dir)
+	{
+		mkdir($self->pg_tde_dir);
+		$self->_setup_default_key;
+		PostgreSQL::Test::Utils::chmod_recursive($self->pg_tde_dir, 0700, 0600);
+	}
+
+	PostgreSQL::Test::Utils::run_log([
+		'pg_tde_change_key_provider',
+		'-D', $self->data_dir,
+		'1664',
+		'global_test_provider',
+		'file',
+		$self->pg_tde_keyfile,
+	]);
+}
+
+sub backup
+{
+	my ($self, $backup_name, %params) = @_;
+	my $backup_dir = $self->backup_dir . '/' . $backup_name;
+
+	mkdir $backup_dir or die "mkdir($backup_dir) failed: $!";
+
+	PostgreSQL::Test::RecursiveCopy::copypath($self->pg_tde_dir, $backup_dir . '/pg_tde');
+	PostgreSQL::Test::Utils::chmod_recursive($backup_dir . '/pg_tde', 0750, 0640);
+
+	$self->SUPER::backup($backup_name, %params);
+}
+
+sub pg_tde_dir
+{
+	my ($self) = @_;
+	return $self->data_dir . '/pg_tde';
+}
+
+sub pg_tde_keyfile
+{
+	my ($self) = @_;
+	return $self->basedir . '/pg_tde_test_keys';
+}
+
+sub _setup_keyfile
+{
+	my ($self) = @_;
+
+	$self->_write_binary_file($self->pg_tde_keyfile, $self->_key_provider_file);
+
+}
+
+sub _setup_default_key
+{
+	my ($self, %params) = @_;
+
+	$self->_write_binary_file(
+		$self->pg_tde_dir . '/1664_providers',
+		$self->_provider_config_file
+	);
+	chmod(0600, $self->pg_tde_dir . '/1664_providers');
```

Comment:
In real code one would use `umask` or `sysopen`, but honestly it does not matter here.

Reply:
Yeah, it just trigggered my autoamtic scanning for security issues.